lib/fs_utils/write.js

```diff
@@ -131,17 +131,13 @@ const write = (fileList, config, joinConfig, optimizers, startTime) => {
       });
   });
 
-  const reduceGenerator = (previousPromise, file) => {
-    return previousPromise.then(() => {
-      return generate(file.path, file.sourceFiles, config, optimizers);
-    }, err => {
-      return Promise.reject(err);
-    });
+  const gen = file => {
+    return generate(file.path, file.sourceFiles, config, optimizers);
   };
 
   return Promise.resolve(changed)
     .then(changedFiles => {
-      return changedFiles.reduce(reduceGenerator, Promise.resolve());
+      return Promise.all(changedFiles.map(gen));
```

Comment:
We need this so that optimizations are actually run in parallel, not in sequence... otherwise, there would be no gain in parallelizing these